src/main/java/net/countercraft/movecraft/combat/features/FlammableOverride.java

```diff
@@ -0,0 +1,186 @@
+        try {
+            for (Material m : flammabilityOverride.keySet()) {
+                if (!revertToVanilla(m))
+                    MovecraftCombat.getInstance().getLogger().warning("Unable to reset " + m.name());
+            }
+        } catch (Exception e) {
+            MovecraftCombat.getInstance().getLogger().info("Failed to disable FlammabilityOverride: ");
+            e.printStackTrace();
+        }
+    }
+
+    private static class FlammabilityNMS {
+
+        public boolean setBlastResistance(Material m, int flammability, int encouragement) {
+            throw new NotImplementedException();
+        }
+
+        protected static void writeField(@NotNull Object fireBlock, @NotNull Object block, String igniteOddsName, String burnOddsName, int flammability, int encouragement)
+                throws IllegalAccessException, NoSuchFieldException {
+            /*for(Field f : fireBlock.getClass().getDeclaredFields()) {
+                f.setAccessible(true);
+                MovecraftCombat.getInstance().getLogger().warning(f.getName() + ": " + f.getType().toString());
+            }*/
+
+            Field igniteOddsField = fireBlock.getClass().getDeclaredField(igniteOddsName);
+            igniteOddsField.setAccessible(true);
+            Object2IntMap igniteOddsMap = (Object2IntMap)igniteOddsField.get(fireBlock);
+            igniteOddsMap.put(block, encouragement);
+
+            Field burnOddsField = fireBlock.getClass().getDeclaredField(burnOddsName);
+            burnOddsField.setAccessible(true);
+            Object2IntMap burnOddsMap = (Object2IntMap)burnOddsField.get(fireBlock);
+            burnOddsMap.put(block, flammability);
+        }
+    }
+
+    private static class FlammabilityNMS_SpigotMappings extends net.countercraft.movecraft.combat.features.FlammableOverride.FlammabilityNMS {
+        @NotNull
+        protected static Class<?> getCraftMagicNumbersClass() throws ClassNotFoundException {
+            String packageName = Bukkit.getServer().getClass().getPackage().getName();
+            String version = packageName.substring(packageName.lastIndexOf('.') + 1);
+            return Class.forName("org.bukkit.craftbukkit." + version + ".util.CraftMagicNumbers");
+        }
+
+        protected static Object getBlockClass(@NotNull Class<?> magicNumbers, Material m)
+                throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+            Method method = magicNumbers.getMethod("getBlock", Material.class);
+            return method.invoke(null, m);
+        }
+
+        public boolean setBlastResistance(Material m, int flammability, int encouragement) {
+            try {
+                Object block = getBlockClass(getCraftMagicNumbersClass(), m);
+                Object fireBlock = getBlockClass(getCraftMagicNumbersClass(), Material.FIRE);
+                writeField(fireBlock, block, "O", "P", flammability, encouragement); // obfuscated explosionResistance
```

Comment:
Comment likely needs updates

Reply:
Yes, left over from copieng the blast resistance override code, i will change it later